### PR TITLE
Vot 113 tokens container

### PR DESCRIPTION
**Pull request by c0rv0s**
# Short description
Shows a container on the /tokens page for the tokens that you own

# How can it be tested
Send some registered tokens to your address then click 'Find Tokens' in the nav bar

# Tracker ID
https://linear.app/aragon/issue/VOT-113/show-your-tokens-container

### CODE REVIEW

**Review by cbrzn — COMMENTED** [review]
Looks awesome! Just left a few comments to improve it

Good work bro

**Comment by cbrzn** [comment]
Also, I feel that we should start doing micro improvements, and I feel that we should create a `services` folder, where maybe we have an `ethereum` folder? (Something like this https://github.com/dOrgTech/homebase-app/tree/develop/src)

And creating the hooks inside of every service like here https://github.com/dOrgTech/homebase-app/blob/develop/src/services/contracts/baseDAO/hooks/useVote.ts

If you'd like to do any improvement feel free to do it, anyways, we will tackle this in any moment

File: lib/constants.ts [hunk 1]
```diff
@@ -14,3 +14,32 @@ export const TRUST_WALLET_BASE_URL =
...
+          {
+              "internalType": "uint256",
+              "name": "",
+              "type": "uint256"
+          }
+      ],
+      "payable": false,
+      "stateMutability": "view",
+      "type": "function"
+  }
```
[the remark this answers is not in the document]
Comment (cbrzn) [reply lib/constants.ts:40]: Actually you can use it from here: https://github.com/vocdoni/bridge-ui/blob/develop/lib/constants.ts#L3

File: lib/hooks/user-tokens.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    const { registeredTokens } = useRegisteredTokens();
+    const { setAlertMessage } = useMessageAlert()
+    const wallet = useWallet<providers.JsonRpcFetchFunc>();
+    
+    const refreshUserTokens = async (tokenAddresses: string[]): Promise<TokenBalance[]> => {
+        if (!wallet?.ethereum || !wallet?.account) {
+            setError('No wallet connected')
+            return []
+        }
+        if (wallet.chainId === GOERLI_CHAINID) setMulticallAddress(wallet.chainId, GOERLI_MULTICALL)
```
[the remark this answers is not in the document]
Comment (c0rv0s) [reply lib/hooks/user-tokens.tsx:41]: not necessary! goerli is the only network that doesnt have the multicall address included already

[the remark this answers is not in the document]
Comment (cbrzn) [reply lib/hooks/user-tokens.tsx:41]: Ohh okok, perfect

File: lib/constants.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (c0rv0s) [reply lib/constants.ts:40]: I'm with you on this but this package doesn't support them yet. I opened a pr on the repo to add support, maybe we keep it like this until the maintainer can publish? he's pretty active

File: pages/tokens/index.tsx [hunk 3]
```diff
@@ -64,7 +66,32 @@ const TokensPage = () => {
...
+              <p>
+                {token?.name || "(loading)"}
+                <br />
+                {token?.totalSupply && <small>Total supply: {token?.totalSupplyFormatted}</small>}
+              </p>
+            </TokenCard>
+          ))}
+      </TokenList>
+
+      <ActiveTokens>Active tokens</ActiveTokens>
```
[the remark this answers is not in the document]
Comment (c0rv0s) [reply pages/tokens/index.tsx:92]: do you mean on line 67? i think we want a grid of user tokens above a grid of all tokens :P

File: lib/constants.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (cbrzn) [reply lib/constants.ts:40]: This makes me sad :(

File: pages/tokens/index.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (cbrzn) [reply pages/tokens/index.tsx:92]: Yah lol 

> i think we want a grid of user tokens above a grid of all tokens :P
That's in the new design correct? If so, I'd say that we can just change the title for now (as the service has been implemented) and in another PR tackle the UI (Sorry if I am wrong)

[the remark this answers is not in the document]
Comment (c0rv0s) [reply pages/tokens/index.tsx:92]: the new design yea, so just one token list of user tokens then?

[the remark this answers is not in the document]
Comment (cbrzn) [reply pages/tokens/index.tsx:92]: I'd say that for now we can just leave it with "My tokens" just to show that the service is already implemented and then make it good looking

**Review by cbrzn — APPROVED** [review]

**Pull request merged**